src/main/scala/io/scalac/wtf/domain/UserService.scala

```diff
@@ -14,14 +16,15 @@ object UserService {
     def pure[A](a: A): DBIO[A] = DBIO.successful(a)
     def flatMap[A, B](fa: DBIO[A])(f: A => DBIO[B]) = fa.flatMap(f)
   }
-
+  
   def createUser(createdUser: User)(implicit ec: ExecutionContext): DBIO[Xor[NonEmptyList[ValidationError], UserId]] = {
     val resT = for {
       validUser <- XorT[DBIO, NonEmptyList[ValidationError], User](User.validateUser(createdUser.email, createdUser.password))
-      userId    <- XorT.right[DBIO, NonEmptyList[ValidationError], UserId](UserRepository.save(validUser))
+      userId    <- XorT.right[DBIO, NonEmptyList[ValidationError], UserId](UserRepository.save(hashUserPassword(validUser)))
     } yield userId
 
     resT.value
   }
 
+  private def hashUserPassword(user: User): User = user.copy(password = user.password.bcrypt)
```

Comment:
If I knew it will block this PR for so long, I wouldn't have asked. 
I would go with a "rich domain" approach: `UserPrototype` -> validation -> `User` -> save to db
But it's not the most important thing